src/create.cpp

```diff
@@ -149,7 +149,110 @@ auto createFromRoot( std::string_view root_, std::string_view indexLocation, boo
+	 *  - ContentRoot (wherever it appears) is relative
+	 *  - DepotBuildConfig/FileMapping/LocalPath is relative
+	 *  - DepotBuildConfig/FileMapping/LocalPath contains files directly under DepotBuildConfig/FileMapping/DepotPath
+	 *    (for example, LocalPath is ".\p2ce\x.txt" and DepotPath is ".\p2ce\")
+	 *  - DepotBuildConfig/FileMapping/Recursive is true when LocalPath is a directory
+	 */
+	Log_Warn( "{}", "The Steam depot config parser is incomplete and fine-tuned to work with Portal 2: Community Edition. Use at your own risk." );
+
+	auto start{ std::chrono::high_resolution_clock::now() };
+	unsigned int configs = 0;
+
+	if (! std::filesystem::exists( configPath ) ) {
+		Log_Error( "Depot config at `{}` does not exist!", configPath );
+		return 1;
+	}
+
+	KV1 configKeyvalues{ sourcepp::fs::readFileText( configPath ) };
+	if ( configKeyvalues.hasChild( "DepotBuildConfig" ) ) {
+		Log_Error( R"(Depot config at `{}` has root key "DepotBuildConfig". Use the config with the root key "AppBuild" instead.)", configPath );
+		return 1;
+	}
+
+	const auto& appBuildConfig = configKeyvalues[ "AppBuild" ];
+	if ( appBuildConfig.isInvalid() ) {
+		Log_Error( "Depot config at `{}` is invalid!", configPath );
+		return 1;
+	}
+
+	auto contentRoot{ std::filesystem::path{ configPath }.parent_path() / appBuildConfig[ "ContentRoot" ].getValue() };
+	const auto& depots = appBuildConfig[ "Depots" ];
+	for ( const auto& depot : depots.getChildren() ) {
+		if ( std::find( depotIDs.begin(), depotIDs.end(), depot.getKey() ) == depotIDs.end() ) {
+			continue;
+		}
+
+		const auto createFromSteamDepotConfig{ [ &configPath, &indexLocation, skipArchives, &defaultExclusions, &contentRoot ]( const KV1Element& depotBuildConfig ) {
+			std::vector<std::string> exclusionRegexes;
+			exclusionRegexes.insert( exclusionRegexes.end(), defaultExclusions.begin(), defaultExclusions.end() );
+			for ( int i = 0; i < depotBuildConfig.getChildCount( "FileExclusion" ); i++ ) {
+				std::string exclusion{ depotBuildConfig( "FileExclusion", i ).getValue() };
+				sourcepp::string::normalizeSlashes( exclusion );
+				if ( exclusion.starts_with( "./" ) )
+					exclusion = exclusion.substr(2);
+
+				exclusionRegexes.emplace_back( globToRegex( exclusion ) );
+			}
+
+			std::vector<std::string> inclusionRegexes;
+			for ( int i = 0; i < depotBuildConfig.getChildCount( "FileMapping" ); i++ ) {
+				std::string inclusion{ depotBuildConfig( "FileMapping", i )[ "LocalPath" ].getValue() };
+				sourcepp::string::normalizeSlashes( inclusion );
+				if ( inclusion.starts_with( "./" ) )
+					inclusion = inclusion.substr(2);
+
+				inclusionRegexes.emplace_back( globToRegex( inclusion ) );
+			}
+
+			createFromRoot(
+				depotBuildConfig.hasChild( "ContentRoot" ) ? ( std::filesystem::path{ configPath }.parent_path() / depotBuildConfig[ "ContentRoot" ].getValue() ).string() : contentRoot.string(),
+				indexLocation,
+				skipArchives,
+				exclusionRegexes,
+				inclusionRegexes,
+				{ },
```

Comment:
This is for the archive exclusion/inclusion regexes right? These should be taken from the CLI options